### DISCUSSION TITLE
How to do a GitHub pull request

### DISCUSSION THREAD

**Discussion by lucky18210** [community/community#29314] [Pull Requests] [answered]
How do I create and/or send a pull request to another repository hosted on GitHub?

**Comment by luck18210** [comment, marked as answer]
To learn how to make a pull request I just followed two separate help pages on Github (linked below as bullet points). The following command line commands are for **Part 1**. **Part 2**, the actual pull request, is done entirely on Github's website.

> $ git clone https://github.com/tim-peterson/dwolla-php.git
> $ cd dwolla-php
> $ git remote add upstream https://github.com/Dwolla/dwolla-php.git
> $ git fetch upstream
> // make your changes to this newly cloned, local repo 
> $ git add .
> $ git commit -m '1st commit to dwolla'
> $ git push origin master

**Part 1**: fork someone's repo: https://help.github.com/articles/fork-a-repo

- click the 'fork' button on the repo you want to contribute to, in this case: Dwolla's PHP repo (Dwolla/dwolla-php)
- get the URL for your newly created fork, in this case: https://github.com/tim-peterson/dwolla-php.git (tim-peterson/dwolla-php)
- type the git clone->cd dwolla-php->git remote->git fetch sequence above to clone your fork somewhere in your computer (i.e., "copy/paste" it to, in this case: third_party TimPeterson$) and sync it with the master repo (Dwolla/dwolla-php)
- make your changes to your local repo
- type the git add->git commit->git push sequence above to push your changes to the remote repo, i.e., your fork on Github (tim-peterson/dwolla-php)

**Part 2**: make pull-request: https://help.github.com/articles/using-pull-requests

- go to your fork's webpage on Github (https://github.com/tim-peterson/dwolla-php)
- click 'pull-request' button
- give pull-request a name, fill in details of what changes you made, click submit button.
- you're done!!

**Reply by lucky18210** [reply]
Thank you so much.

**Reply by luck18210** [reply]
You are welcome
Please let me know if you need my help.
Regards

**Reply by lucky18210** [reply]
OK, Thank you.

**Comment by lucky18210** [comment]
It doesn't work, but I can't find the cause.

**Reply by luck18210** [reply]
What error are you getting?

**Reply by lucky18210** [reply]
I am not clear
I'll give you the specifics in a moment.

**Reply by luck18210** [reply]
I see
